pkg/collector/corechecks/snmp/internal/report/report_metrics.go

```diff
@@ -37,12 +40,32 @@ type MetricSample struct {
 	options    profiledefinition.MetricsConfigOption
 }
 
+// InterfaceRate tracks the interface's current ifSpeed and last seen sample to generate rate with
+type InterfaceRate struct {
```

Comment:
Naming feedback: It seems `InterfaceRate` doesn't fully describe what we are storing here. Maybe `BandwithUsage` is more accurate?

Same feedback for naming of `InterfaceRateMap`.

Reply:
i agree! i didn't think too hard about the name when i was working on it, but refactoring to something like `BandwidthUsage` i think would make a lot more sense